### DISCUSSION TITLE
Why does every midi track begin with a rest?

### DISCUSSION THREAD

**Discussion by jinnandtonic** [mido/mido#439] [Q&A] [answered]
Trying the `create_midifile.py` script and the track always begins with a rest the length of `delta`.  Why won't the notes start at the beginning of the file?

**Comment by rdoursenaud** [comment, marked as answer]
Because this is sample code and thus is very naive.
To achieve what you want, you’d have to create a special case for the first loop iteration and set the `time` property of that first message to `0`.
